harvester/harvester
# Serve Harvester APIs as Rancher aggregation APIs

Related issue: https://github.com/harvester/harvester/issues/922
Related PR: https://github.com/harvester/harvester-installer/pull/91

The PR includes the following changes:
1. Update dependencies to comply with Rancher 2.6 master
2. Update ConfigureAPIUI  function so that we can use upstream rancher/apiserver instead of the forked one.
3. Create APIService resource on start
4. Serve APIs under /v1/harvester
5. Adapt to Rancher vendor changes

**Test plan:**
1. Install Harvester. Set the rancher-enabled setting to true. Check that Harvester APIs are served under both `<Rancher_URL>/k8s/clusters/local/v1/harvester` and `<Rancher_URL>/v1/harvester`
2. Install Harvester. Install an external Rancher(master-head). Import the Harvester cluster to the external Rancher. Check that Harvester APIs are served under `<Rancher_URL>/k8s/clusters/<cluster-id>/v1/harvester`
3. Do sanity check on the extension Harvester APIs. Check that CRUD virtualmachines/calling VM actions works.
4. Check authn/z on the APIs. Create Rancher users with different roles(admin,cluster member, project owner/member,readonly, etc.) and check that their access to the APIs comply with their permissions.
5. Check that basic operations in the UI are not broken.
